[WIP] Detail the use of Starting Balance transactions with bank sync (#596)
actualbudget/docs: docs/advanced/bank-sync.md

* GoCardless (European Banks)
* SimpleFIN Bridge (North American Banks)

## Best practices

### Starting Balance Transactions

> If you're planning on setting up bank integration on a new account in Actual, its a good idea to have your account balances already set before hand ("Starting Balance Transaction"). Doing so allows you to have an accurate budget straight from the start. Having a Starting Balance transaction also tells Actual to only sync transactions from the date of the transaction forward.

Reviewer: ```suggestion
If you plan on setting up bank integration on a new account in Actual, you should set your account balances beforehand ("Starting Balance Transaction"). Doing so allows you to have an accurate budget straight from the start. Having a Starting Balance transaction also tells Actual to only sync transactions from the Starting Balance transaction date forward.
```
